scripts/api/createService.ts

```diff
@@ -0,0 +1,133 @@
+  console.log('activating version 1')
+  await activateVersion(createResponse.id, 1)
+  console.log('Version 1 activated, Service created!')
+  return client.getServiceDetail({ service_id: createResponse.id })
+}
+
+async function createOrigin(serviceId: string, domain: string) {
+  console.log('Creating default origin')
+  const client = createClient('backend')
+  await client.createBackend({
+    service_id: serviceId,
+    version_id: 1,
+    address: process.env.DEFAULT_ORIGIN,
+    name: 'default-backend',
+    port: 80,
+    override_host: domain,
+    use_ssl: false,
+  })
+  console.log('Default origin created')
+}
+
+async function createDomain(domain: string, serviceId: string) {
+  console.log('Creating domain', domain)
+  const domainClient = createClient('domain')
+  await domainClient.createDomain({
+    version_id: 1,
+    name: domain,
+    service_id: serviceId,
+  })
+  await domainClient.createDomain({
+    version_id: 1,
+    name: `fpjs-fastly-${domain.split('.')[0]}.edgecompute.app`,
+    service_id: serviceId,
+  })
+}
+
+async function linkConfigStore(service_id: string, version_id: number, resource_id: string) {
+  return createClient('resource').createResource({
+    service_id,
+    version_id,
+    resource_id,
+    name: `${service_id}-config-store-link-${resource_id}`,
+  })
+}
+
+async function createConfigStore() {
+  console.log('Creating config store')
+  const configStoreClient = createClient('configStore')
+  const configStoreItemClient = createClient('configStoreItem')
+  let configStore
+  try {
+    configStore = await configStoreClient.createConfigStore({
+      name: CONFIG_STORE_NAME,
+    })
+  } catch (_) {
+    const stores = await configStoreClient.listConfigStores()
+    return stores.find((t: any) => t.name === CONFIG_STORE_NAME)
+  }
+  await configStoreItemClient.createConfigStoreItem({
+    config_store_id: configStore.id,
+    item_key: 'GET_RESULT_PATH',
+    item_value: process.env.GET_RESULT_PATH ?? 'result',
+  })
+  await configStoreItemClient.createConfigStoreItem({
+    config_store_id: configStore.id,
+    item_key: 'AGENT_SCRIPT_DOWNLOAD_PATH',
+    item_value: process.env.AGENT_SCRIPT_DOWNLOAD_PATH ?? 'agent',
+  })
+  await configStoreItemClient.createConfigStoreItem({
+    config_store_id: configStore.id,
+    item_key: 'PROXY_SECRET',
+    item_value: 'secret',
+  })
+  await configStoreItemClient.createConfigStoreItem({
+    config_store_id: configStore.id,
+    item_key: 'OPEN_CLIENT_RESPONSE_ENABLED',
+    item_value: 'false',
+  })
+
+  return configStore
+}
+
+async function createBackends(service_id: string, version_id: number) {
+  const client = createClient('backend')
+  await client.createBackend({
+    service_id,
+    version_id,
+    address: process.env.FPJS_BACKEND_URL ?? 'api.fpjs.io',
+    name: 'fpjs',
+    port: 443,
+  })
+  await client.createBackend({
+    service_id,
+    version_id,
+    address: process.env.FPCDN_BACKEND_URL ?? 'fpcdn.io',
```

Comment:
nitpick: I'd go with the name `FPCDN_BACKEND_URL` instead of `FPCDN_URL` because it is not a traditional backend, but this is not _that_ important.

Reply:
This commit refactors it: https://github.com/fingerprintjs/fingerprint-pro-fastly-proxy-integration-example/pull/7/commits/5ec6135dc3a6a23be6413f407fdefad2dde93aac